engine/src/main/scala/cromwell/engine/workflow/tokens/TokenPool.scala

```diff
@@ -30,11 +30,13 @@ object TokenPool {
 
 final case class SizedTokenPool(pool: Set[JobExecutionToken], override val currentLoans: Set[JobExecutionToken]) extends TokenPool {
 
-  override def pop(): TokenPoolPop = pool.toList match {
-    case head :: tail => TokenPoolPop(SizedTokenPool(tail.toSet, currentLoans + head), Some(head))
-    case Nil => TokenPoolPop(SizedTokenPool(Set.empty, currentLoans), None)
+  override def pop(): TokenPoolPop = if (pool.isEmpty) {
+    TokenPoolPop(SizedTokenPool(Set.empty, currentLoans), None)
+  } else {
+    TokenPoolPop(SizedTokenPool(pool.tail, currentLoans + pool.head), pool.headOption)
```

Comment:
Totally ToL: Is it guaranteed that multiple calls to Set.tail,Set.head (and headOption) will always return the same value ? (since there is no ordering)